agent/engine/docker_client.go

```diff
@@ -1134,6 +1145,85 @@ func (dg *dockerGoClient) removeVolume(ctx context.Context, name string) error {
 	return nil
 }
 
+type listPluginsResponse struct {
+	Plugins []docker.PluginDetail
+	Error   error
+}
+
+// ListPluginsWithFilters currently is a convenience method as go-dockerclient doesn't implement fitered list. When we or someone else submits
+// PR for the fix we will refactor this to pass in the fiters.
+func (dg *dockerGoClient) ListPluginsWithFilters(enabledFilter bool, capabilityFilter []string, timeout time.Duration) ([]string, error) {
```

Comment:
nit: `enabled` and `capabilities` might be enough here for param names.

Reply:
Fixed